gobblin-api/src/main/java/gobblin/configuration/State.java

```diff
@@ -98,6 +102,19 @@ public void setProp(String key, Object value) {
   }
 
   /**
+   * Appends the input value to a list property that can be retrieved with {@link #getPropAsList}.
+   * @param key property key
+   * @param value property value (should not include commas, which are used as separators)
```

Comment:
The separator can't really be changed because getPropAsList is an existing API that relies on separator being always ",". I did think about the fact that values could have commas. Ideally we would want to escape and unescape the strings, but it is not particularly easy.
Two options: either spend the time doing the escaping, or just specify in the doc that if values have commas they will be parsed incorrectly.